packages/components/src/components/form/Range/Range.tsx

```diff
@@ -58,4 +49,76 @@ const Input = styled.input<Pick<RangeProps, 'thumbStyle' | 'trackStyle'>>`
     }
 `;
 
-export const Range = (props: RangeProps) => <Input type="range" {...props} />;
+const LabelsWrapper = styled.div<{ count: number; width?: number }>`
+    display: grid;
+    grid-template-columns: ${({ count, width }) =>
+        count && width && `repeat(${count}, ${width}px)`};
+    justify-content: space-between;
+`;
+
+const Label = styled.p<{ width?: number }>`
+    position: relative;
+    justify-self: center;
+    padding-top: 2px;
+    min-width: ${({ width }) => `${width}px`};
+    text-align: center;
+    font-size: ${variables.FONT_SIZE.TINY};
+    font-weight: ${variables.FONT_WEIGHT.MEDIUM};
+    opacity: 0.5;
+    cursor: pointer;
+`;
+
+export interface RangeProps {
+    className?: string;
+    max?: number;
+    min?: number;
+    step?: string;
+    onChange: React.ChangeEventHandler<HTMLInputElement>;
+    thumbStyle?: CSSObject;
+    trackStyle?: CSSObject;
+    value: number;
+    labels?: (string | number)[];
+    onLabelClick?: (value: number) => void;
+}
+
+export const Range = ({ labels, onLabelClick, ...props }: RangeProps) => {
+    const [laseElWidth, setLaseElWidth] = useState<number>();
+
+    const lastLabelRef = useRef<HTMLParagraphElement>(null);
+
+    useLayoutEffect(() => {
```

Comment:
This is a nice improvement, but I think there is a simpler way to achieve the same result, without using refs and effects. The grid could have cells of equal width. Each cell would contain a clickable element justified to the center of the cell. First and last cell could be aligned individually.